app/controllers/registration_controller.rb

```diff
@@ -85,51 +86,19 @@ def ensure_lane_exists
   end
 
   def update
-    guests = params[:guests]
-    status = params.dig('competing', 'status')
-    comment = params.dig('competing', 'comment')
-    event_ids = params.dig('competing', 'event_ids')
-    admin_comment = params.dig('competing', 'admin_comment')
-    waiting_list_position = params.dig('competing', 'waiting_list_position')
-
-    begin
-      registration = Registration.find("#{@competition_id}-#{@user_id}")
-      old_status = registration.competing_status
-      if old_status == 'accepted' && status != 'accepted'
-        Registration.decrement_competitors_count(@competition_id)
-      elsif old_status != 'accepted' && status == 'accepted'
-        Registration.increment_competitors_count(@competition_id)
-      end
-      updated_registration = registration.update_competing_lane!({ status: status, comment: comment, event_ids: event_ids, admin_comment: admin_comment, guests: guests, waiting_list_position: waiting_list_position })
-
-      render json: { status: 'ok', registration: {
-        user_id: updated_registration['user_id'],
-        guests: updated_registration['guests'],
-        competing: {
-          event_ids: updated_registration.registered_event_ids,
-          registration_status: updated_registration.competing_status,
-          registered_on: updated_registration['created_at'],
-          comment: updated_registration.competing_comment,
-          admin_comment: updated_registration.admin_comment,
-          waiting_list_position: updated_registration.competing_waiting_list_position,
-        },
-      } }
-    rescue Dynamoid::Errors::Error => e
-      puts e
-      Metrics.registration_dynamodb_errors_counter.increment
-      render json: { error: "Error Updating Registration: #{e.message}" },
-             status: :internal_server_error
-    end
+    render json: { status: 'ok', registration: process_update(params) }
+  rescue StandardError => e
```

Comment:
what happened to rescuing a specific error?

Reply:
Good catch, no idea - fixed